src/transformers/generation/candidate_generator.py

```diff
@@ -319,6 +324,15 @@ def get_candidates(self, input_ids: torch.LongTensor) -> Tuple[torch.LongTensor,
                 if start_idx < end_idx:
                     chosen_ids = input_ids[0, start_idx:end_idx]
                     match_found = True
+
+                    # remove remaining candidate ids if an "eos" token is found, otherwise the target model may
+                    # accept eos and the rest as valid, thus not stopping generation after "eos"
+                    # NOTE: below code is written based on the fact that assisted decoding supports only bs=1
```

Comment:
This will work in all cases, if the input contains and eos or doesn't. It not always the case that an eos is chosen as candidate, that depends on the overlap between prompt and the generated part

Reply:
Ah, sorry, I meant re batch size 1, can we assume this will always hole?

Reply:
Yes, the current impl doesn;t support more than 1. If anyone adds support, this and other parts of postprocessing will have to be modified